### PR TITLE
[SPARK-15198][SQL] Support for pushing down filters for boolean types in ORC data source

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

It seems ORC supports all the types in  ([`PredicateLeaf.Type`](https://github.com/apache/hive/blob/e085b7e9bd059d91aaf013df0db4d71dca90ec6f/storage-api/src/java/org/apache/hadoop/hive/ql/io/sarg/PredicateLeaf.java#L50-L56)) which includes boolean types. So, this was tested first.

This PR adds the support for pushing filters down for `BooleanType` in ORC data source.

This PR also removes `OrcTableScan` class and the companion object, which is not used anymore.
## How was this patch tested?

Unittest in `OrcFilterSuite` and `OrcQuerySuite`.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
Let me please cc @liancheng and also @tedyu who suggested this change.

**Comment by HyukjinKwon** [comment]
Please excuse my ping, @liancheng

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
retest this please

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcQuerySuite.scala [hunk 1]
```diff
@@ -451,4 +451,17 @@ class OrcQuerySuite extends QueryTest with BeforeAndAfterAll with OrcTest {
...
+
+  test("SPARK-15198 Support for pushing down filters for boolean types") {
+    withSQLConf(SQLConf.ORC_FILTER_PUSHDOWN_ENABLED.key -> "true") {
+      val data = (0 until 10).map(_ => (true, false))
+      withOrcFile(data) { file =>
+        val actualTwo = spark.read.orc(file).where("_2 == true")
+
+        // ORC filter should be applied and the total count should be less than
+        // the original data.
+        assert(stripSparkFilter(actualTwo).count() < data.length)
```
Comment (liancheng) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcQuerySuite.scala]: If I understand this correctly, `.count()` here should be just zero?

Comment (liancheng) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcQuerySuite.scala]: I mean, if it is, we probably should just assert that it's zero.

**Comment by liancheng** [comment]
Sorry for the late review, this LGTM except for one minor comment on the test case. Thanks for fixing it!

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcQuerySuite.scala (hunk 1 above)
Comment (HyukjinKwon) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcQuerySuite.scala]: Ah, yes it is because there would be single stripe in single file!

**Comment by HyukjinKwon** [comment]
Thank you so much.

**Comment by HyukjinKwon** [comment]
Hi @liancheng, I guess you missed this. I will cc in case.

**Comment by HyukjinKwon** [comment]
Hi @liancheng ! Could you take a quick look please?

**Comment by HyukjinKwon** [comment]
ping @liancheng ...

**Comment by HyukjinKwon** [comment]
@yhuai Could you please take a look?

**Comment by liancheng** [comment]
LGTM, merging to master. Sorry for leaving this PR for so long...

**Comment by HyukjinKwon** [comment]
No problem! thank you!

**Pull request closed without merging**